### PR TITLE
Updates macOS support after #62

### PR DESCRIPTION
**Pull request by wenkokke**
- Setup CI for macOS
- Remove outdated information with regards to macOS builds from README

### CODE REVIEW

**Comment by wenkokke** [comment]
Adding WASM tests, since I believe that they still have the infinite loop in parsing HLS... somehow?

**Comment by wenkokke** [comment]
Currently a draft as 'm attempting to add in wasm tests

File: script/tree-sitter-parse.js [hunk 1]
```diff
@@ -13,7 +13,7 @@ Parser.init().then(() => {
     const parser = new Parser;
     parser.setLanguage(Haskell);
     for (let i = 2; i < process.argv.length - 1; i++) {
-      const fileName = process.argv[2]
+      const fileName = process.argv[i]
```
Comment (tek) [inline script/tree-sitter-parse.js:16]: :grimacing: :sweat_smile:

Comment (wenkokke) [reply script/tree-sitter-parse.js:16]: Yeah, I noticed. It's fixed locally.

**Comment by wenkokke** [comment]
@tek This is ready to be merged

**Pull request merged**